docs/source/docs/configuration.md

```diff
@@ -2,19 +2,459 @@
+loader which one to use. We can use the `sample_label` argument to indicate which method
+to use. Pass one of "sample", for random sampling, or "all" to use all the available
+labels:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  type: "paired" # one of "paired", "unpaired" or "grouped"
+  labeled: true
+  sample_label: "sample" # one of "sample", "all" or None
+```
+
+In the case the `labeled` argument is false, the sample_label is unused, but still must
+be passed. Additionally, if the tensors in the files only have one label, irregardles of
+the `sample_label` argument, the data loader will only pass the one label to the
+network.
+
+##### Paired
+
+- `moving_image_shape`: (list, tuple) len 3, corresponding to (dim1, dim2, dim3) of the
+  3D moving image.
+- `fixed_image_shape`: (list, tuple) len 3, corresponding to (dim1, dim2, dim3) of the
+  3D fixed image.
+
+##### Unpaired
+
+- `image_shape`: (list, tuple) len 3, corresponding to (dim1, dim2, dim3) of the 3D
+  image.
+
+##### Grouped
+
+- `intra_group_prob`: float, between 0 and 1. Passing 0 would only generate inter-group
+  samples, and passing 1 would only generate intra-group samples.
+- `sample_label`: method for sampling the labels "sample", "all" `intra_group_option`:
+  str, "forward", "backward, or "unconstrained"
+- `sample_image_in_group`: bool, if true, only one image pair will be yielded for each
+  group, so one epoch has num_groups pairs of data, if false, iterate through this
+  loader will generate all possible pairs.
+- `image_shape`: (list, tuple) len 3, corresponding to (dim1, dim2, dim3) of the 3D
+  image.
+
 See the [dataset loader configuration](dataset_loader.html) for more details.
 
 ## Train section
 
 The `train` section defines the neural network training hyper-parameters, by specifying
-subsections, `model`, `loss`, `optimizer`, `preprocess` and other training
-hyper-parameter, including `epochs` and `save_period`. See an
-[example configuration](https://github.com/DeepRegNet/DeepReg/blob/main/config/unpaired_labeled_ddf.yaml),
-with comments on the available options in each subsection.
+subsections, `method`, `backbone`, `loss`, `optimizer`, `preprocess` and other training
+hyper-parameters, including `epochs` and `save_period`.
+
+### Method - required
+
+The `method` argument defines the registration type. It must be a string type, one of
+"ddf", "dvf", "conditional", which are the currently supported registration methods.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+```
+
+### Backbone - required
+
+The `backbone` section defines the backbone network for the registration. This section
+has several subsections. The `name` and `num_channel_initial` are global to all backbone
```

Comment:
only `name` is required, all other params will be passed to the backbone class directly, https://github.com/DeepRegNet/DeepReg/blob/ba27961deb0a3d1721747e03ae5a25ef49d988ea/deepreg/model/network/util.py#L56

maybe link the doc to some code can be helpful?

Reply:
Yeah, so I was going off the implem prior to the registry. How would this work then? We still need to pass  those args into the config right?

Reply:
@mathpluscode I am trying to resolve this now but I still don't understand how the config changes from the Registry code. So, do we have something like this?

```yaml
train:
  method: "ddf" # One of ddf, dvf, conditional
  name:
    backbone: "unet"  # One of unet, local, global
    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
    depth: 3
    pooling: False
    concat_skip: True
```

where do we put the class name? under name? I am not sure how to proceed at the moment.

Reply:
@YipengHu  see this discussion

Reply:
oh sorry I missed this msg, is it clear now for you?

Reply:
it should be 
```
train:
  method: "ddf" # One of ddf, dvf, conditional
  backbone:
    name: "unet"  # One of unet, local, global
    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
    depth: 3
    pooling: False
    concat_skip: True
```

Reply:
my bad, i should provide the example for u

Reply:
Solved.